src/realm/obj.cpp

```diff
@@ -761,10 +770,87 @@ void out_mixed(std::ostream& out, const Mixed& val)
     }
 }
 
+void out_mixed_xjson(std::ostream& out, const Mixed& val)
+{
+    if (val.is_null()) {
+        out << "null";
+        return;
+    }
+    switch (val.get_type()) {
+        case type_Int:
+            out << "{\"$numberLong\": \"";
+            out << val.get<Int>();
+            out << "\"}";
+            break;
+        case type_Bool:
+            out << (val.get<bool>() ? "true" : "false");
+            break;
+        case type_Float:
+            out << "{\"$numberDouble\": \"";
+            out_floats<float>(out, val.get<float>());
+            out << "\"}";
+            break;
+        case type_Double:
+            out << "{\"$numberDouble\": \"";
+            out_floats<double>(out, val.get<double>());
+            out << "\"}";
+            break;
+        case type_String: {
+            out << "\"";
+            out_string(out, val.get<String>());
+            out << "\"";
+            break;
+        }
+        case type_Binary: {
+            out << "{\"$binary\": {\"base64\": \"";
+            out_binary(out, val.get<Binary>());
+            out << "\", \"subType\": \"00\"}}";
+            break;
+        }
+        case type_Timestamp: {
+            out << "{\"$date\": {\"$numberLong\": \"";
+            auto ts = val.get<Timestamp>();
+            int64_t timeMillis = ts.get_seconds() * 1000 + ts.get_nanoseconds() / 1000000;
+            out << timeMillis;
+            out << "\"}}";
+            break;
+        }
+        case type_Decimal:
+            out << "{\"$numberDecimal\": \"";
+            out << val.get<Decimal128>();
+            out << "\"}";
+            break;
+        case type_ObjectId:
+            out << "{\"$oid\": \"";
```

Comment:
i might be missing it, but I don't see this one getting used in the `.json` test files?

Reply:
Hmm, you are correct that these are missing from test coverage for the json / xjson output. I will update the tests to include these